pkg/broker/awsbroker.go

```diff
@@ -241,127 +211,130 @@ func (b *AwsBroker) ValidateBrokerAPIVersion(version string) error {
-								"type":       "object",
-								"properties": propsForCreate,
-								"$schema":    "http://json-schema.org/draft-06/schema#",
-							},
-						},
-					},
+	params := cfnParamsToOsb(sd)
+	for k, p := range sd.Metadata.Spec.ServicePlans {
+		planid := uuid.NewV5(db.Accountuuid, "service__"+sd.Metadata.Spec.Name+"__plan__"+k).String()
+		plan := osb.Plan{
+			ID:          planid,
+			Name:        k,
+			Description: p.Description,
+			Free:        &f,
+			Bindable:    &t,
+			Metadata: map[string]interface{}{
+				"cost":            p.Cost,
+				"displayName":     p.DisplayName,
+				"longDescription": p.LongDescription,
+			},
+			Schemas: &osb.Schemas{ServiceInstance: &osb.ServiceInstanceSchema{}},
+		}
+		propsForCreate := make(map[string]interface{})
+		for nk, nv := range nonCfnParamDefs {
+			propsForCreate[nk] = nv
+		}
+		propsForUpdate := make(map[string]interface{})
+		requiredForCreate := make([]string, 0)
+		requiredForUpdate := make([]string, 0)
+		prescribed := make(map[string]string)
+		var openshiftFormCreate []OpenshiftFormDefinition
+		var openshiftFormUpdate []OpenshiftFormDefinition
+		for paramName, paramValue := range params {
+			include := true
+			for planParam, planValue := range p.ParameterValues {
+				if planParam == paramName {
+					include = false
+					prescribed[planParam] = planValue
 				}
-				if len(requiredForCreate) > 0 {
-					// Cloud Foundry does not allow "required" to be an empty slice
-					plan.Schemas.ServiceInstance.Create.Parameters.(map[string]interface{})["required"] = requiredForCreate
+			}
+			required := false
+			if paramValue.(map[string]interface{})["required"] != nil {
+				required = paramValue.(map[string]interface{})["required"].(bool)
+			}
+			if include {
+				openshiftFormCreate = openshiftFormAppend(openshiftFormCreate, paramName, paramValue.(map[string]interface{}))
+				createParam := paramValue.(map[string]interface{})
+				delete(createParam, "required")
+				delete(createParam, "display_group")
+				propsForCreate[paramName] = createParam
+				if required {
+					requiredForCreate = append(requiredForCreate, paramName)
 				}
-				if len(propsForUpdate) > 0 {
-					plan.Schemas.ServiceInstance.Update = &osb.InputParametersSchema{
-						Parameters: map[string]interface{}{
-							"type":       "object",
-							"properties": propsForUpdate,
-							"$schema":    "http://json-schema.org/draft-06/schema#",
-						},
-					}
-					if len(requiredForUpdate) > 0 {
-						// Cloud Foundry does not allow "required" to be an empty slice
-						plan.Schemas.ServiceInstance.Update.Parameters.(map[string]interface{})["required"] = requiredForUpdate
+				if stringInSlice(paramName, sd.Metadata.Spec.UpdatableParameters) {
+					openshiftFormUpdate = openshiftFormAppend(openshiftFormUpdate, paramName, paramValue.(map[string]interface{}))
+					updateParam := paramValue.(map[string]interface{})
+					delete(updateParam, "required")
+					delete(updateParam, "display_group")
```

Comment:
We may want to delete `default` from the update schema, since we don't use the template default; we default to the value used at provision-time.